backend/resources/hacknight.py

```diff
@@ -54,3 +48,35 @@ def get(self, id):
             {"hacknights": hacknight_schema.dump(Hacknight.query.get_or_404(id))},
             HTTPStatus.OK,
         )
+
+
+class HacknightParticipants(Resource):
+    @jwt_required
+    def post(self, id):
+        hacknight = Hacknight.query.get_or_404(id)
+        participants = [participant.id for participant in hacknight.participants]
+
+        json_data = request.get_json(force=True)
+        ids_schema = Schema.from_dict({"participants_ids": fields.List(fields.Int())})
+        try:
+            data = ids_schema().load(json_data)
+        except ValidationError as err:
+            return err.messages, HTTPStatus.UNPROCESSABLE_ENTITY
+
+        new_participants = [
+            id for id in data["participants_ids"] if id not in participants
```

Comment:
Probably nitpick but maybe could you rename `id` on `_id` or something similar to avoid overwrite python id function.